### PR TITLE
Reset membership

### PR DESCRIPTION
**Pull request by chriskrogh**
## What's changed
Added cron job to 
1. store member info in a google sheet
2. reset membership list

Here's how I derived the cron expression: https://crontab.guru/#0_0_1_JAN,MAY,SEP_*
We're running this job once on the 1st of Jan, May, Sept

### Membership resets
We just set everyone's membership in the db to `EXPIRED`. Here's a sc of 2 users' membership status after the script was run

One was paid and the other was unpaid

Screen Shot 2020-06-21 at 10 30 45 AM

### Export members
Adds a new sheet to the Google Spreadsheet Doc for each term (each time the script runs) and then exports members to that sheet.

## Notes
Did some manual tests and the job works okay.

### CODE REVIEW

File: src/jobs/membership.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  GoogleSpreadsheet,
+  WorksheetBasicProperties,
+} from 'google-spreadsheet';
+import GoogleCreds from '../config/acs-web-5aaaf-768e669e2d39.json';
+
+const getTerm = () => {
+  const date = new Date();
+  const month = date.getMonth();
+  if (month === 1) {
+    return 'Fall' + (date.getFullYear() - 1);
```
Comment (jtaaa) [inline src/jobs/membership.ts:14]: Wouldn't `month === 1` correspond to winter? And `date.getMonth()` is zero-based so 1 is February

File: src/jobs/membership.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  '1Kn3Fc1vmi2dd9I645BxBQZMXXNvIJi1zEGnBiHuHDtk',
+);
+
+cron.schedule('0 0 1 JAN,MAY,SEP *', async function () {
+  try {
+    await doc.useServiceAccountAuth(GoogleCreds);
+    await doc.loadInfo();
+    const newSheet = await doc.addSheet({
+      title: getTerm(),
+    } as WorksheetBasicProperties);
```
Comment (jtaaa) [inline src/jobs/membership.ts:32]: How come we need a cast here? btw the PR to definitely typed is approved and I think just waiting for another review by the maintainer of `google-spreadsheet`.

File: src/jobs/membership.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        { membershipStatus: MEMBERSHIP_STATUS.UNPAID },
+      ],
+    });
+    members.forEach(async (member) => {
+      const { name, email, watIAMUserId, membershipStatus } = member;
+      const newRow = { name, email, watIAMUserId, membershipStatus };
+      await newSheet.addRow(newRow);
+      Object.assign(member, {
+        membershipStatus: MEMBERSHIP_STATUS.EXPIRED,
+      });
```
Comment (jtaaa) [inline src/jobs/membership.ts:45]: How come we don't just do `member.membershipStatus = MEMBERSHIP_STATUS.EXPIRED` here?

Comment (chriskrogh) [reply src/jobs/membership.ts:45]: I **honestly** can't remember why I did this🤣 I'll use `member.membershipStatus = MEMBERSHIP_STATUS.EXPIRED`

File: src/jobs/membership.ts (hunk 2 above)
Comment (chriskrogh) [reply src/jobs/membership.ts:32]: Good news! https://github.com/theoephraim/node-google-spreadsheet/pull/287#issuecomment-648273964
I'll make a PR upgrading the package and see if that resolves this

File: src/jobs/membership.ts (hunk 1 above)
Comment (chriskrogh) [reply src/jobs/membership.ts:14]: I think we run the job at the end of the term. 
1st of Jan runs for Fall
1st of May runs for Winter
1st of Sept runs for Spring
lmk if this is okay or if I should leave a comment in the code

Comment (jtaaa) [reply src/jobs/membership.ts:11]: Fine by me. Still concerned about `month === 1` being February tho

Comment (chriskrogh) [reply src/jobs/membership.ts:11]: ¡¡¡!!! 

Thanks for the catch!!!

**Pull request merged**